lib/tessel/deployment/rust.js

```diff
@@ -33,32 +37,133 @@ var exportables = {
         program
       };
     },
-    shell: () => {
+    shell: (opts) => {
       return tags.stripIndent `
         #!/bin/sh
-        exec /app/remote-script/rust_executable
+        exec /app/remote-script/${opts.resolvedEntryPoint}
       `;
     },
   },
   lists: lists,
 };
 
-exportables.preBundle = function() {
-  return Promise.resolve();
+// The Rust prebundle step just updated the resolvedEntryPoint
+// property with the Rust binary name
+exportables.preBundle = function(opts) {
+  return new Promise((resolve, reject) => {
+    // Get details of the project
+    var details = exportables.meta.checkConfiguration(opts.target);
+
+    // If it was unable to fetch the details
+    if (typeof details !== 'object') {
+      // Abort the deploy
+      return reject('Unable to parse Cargo.toml');
+    } else {
+      // Update the resolved entry point to the program name
+      opts.resolvedEntryPoint = details.program;
+      // Continue with the deploy
+      resolve();
+    }
+  });
 };
 
-exportables.tarBundle = function() {
-  // This must implement a Promise that resolves with a Buffer
-  // that represents a DIRECTORY containing the compiled Rust
-  // executable.
-  return Promise.resolve(new Buffer([0xFF]));
+// This must implement a Promise that resolves with a Buffer
+// that represents a DIRECTORY containing the compiled Rust
+// executable.
+exportables.tarBundle = function(opts) {
+  return exportables.remoteRustCompilation(opts);
 };
 
-exportables.preRun = function(tessel) {
-  return new Promise((resolve) => {
-    return tessel.connection.exec(commands.chmod('+x', `${Tessel.REMOTE_RUN_PATH}rust_executable`), {}, () => resolve());
-  });
+exportables.preRun = function(tessel, opts) {
+  return tessel.simpleExec(commands.chmod('+x', `${Tessel.REMOTE_RUN_PATH}${opts.resolvedEntryPoint}`));
 };
 
+exportables.remoteRustCompilation = (opts) => {
+  log.info('Compiling Rust code remotely...');
+  return new Promise(function(resolve, reject) {
+
+    // Save our incoming compiled buffer to this array
+    var buffers = [];
+
+    // Parse out the components of the remote rust compilation server
+    if (!opts.rustCC.startsWith('http://')) {
+      // Without an HTTP protocol definition, url.parse will fail
+      opts.rustCC = 'http://' + opts.rustCC
```

Comment:
You'll need to add a semicolon here to pass the linting tests (if only we had the crushing stringency of the Rust compiler!).